dashboard/app/models/pd/workshop_facilitator_daily_survey.rb

```diff
@@ -89,5 +89,14 @@ def self.create_placeholder!(user_id:, day:, pd_session_id:, facilitator_id:, fo
         submission_id: submission_id,
       )
     end
+
+    # @override
+    def duplicate?
+      # See if this user already has a submission for this workshop, session, facilitator, & form.
+      # Note: this duplicate record would fail the uniqueness validation
+      self.class.where(
+        attributes.slice(:user_id, :pd_workshop_id, :pd_session_id, :facilitator_id, :form_id)
```

Comment:
I think workshop_id is redundant here if you have session_id

Reply:
Though actually it shouldn't be for the facilitator form, because we don't support day 0. We always have a session. 

In any case, I'm again mirroring the [uniqueness validation](https://github.com/code-dot-org/code-dot-org/blob/f8bcb7276ae87f4f8a9fc52ea85149b54e6f9e5f/dashboard/app/models/pd/workshop_facilitator_daily_survey.rb#L35), which itself does appear to be redundant. However the table definition and model both allow nil session id. I'd rather not mess with it now, but we can address this as a cleanup later.